### PR TITLE
Automated cherry pick of #1853: Move pullsecret creation right after the namespace is created

### PR DESCRIPTION
**Pull request by doublek**
Cherry pick of #1853 on release-v1.26.

#1853: Move pullsecret creation right after the namespace is created

### CODE REVIEW

**Review by Brian-McM — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
